### PR TITLE
182015 Add custom login and upload to OpenSPP server

### PR DESCRIPTION
**Pull request by renceInbox**
Login 
- [x] Accepts username and password, check the credentials from OpenSPP API 

Upload
- [x] Uploaded files should be saved to OpenSPP, dont save to file system

### CODE REVIEW

File: filesystems.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from pyftpdlib.filesystems import AbstractedFS
+
+
+class OpenSPPFS(AbstractedFS):
+    def open(self, filename: str, mode: str) -> TextIO:
+        dir_tree = filename.split("/")
+        new_filename = "_".join(dir_tree)
+        # Use mkdtemp, so it persist and should be removed on
+        # OpenSPPFTPHandler.on_file_received after uploading to OpenSPP
+        temp_dir = tempfile.mkdtemp()
```
Comment (jeremi) [inline filesystems.py:13]: Can we use in-memory file?

File: local.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    build: .
+    container_name: local_ftp
+    ports:
+      - "8002:8002"
+      - "3000:3000"
+      - "3001:3001"
+      - "3002:3002"
+      - "3003:3003"
+      - "3004:3004"
+      - "3005:3005"
```
Comment (jeremi) [inline local.yml:15]: Can we use port-range instead?

**Review by jeremi — CHANGES_REQUESTED** [review]
please restructure the repo following https://kennethreitz.org/essays/2013/01/27/repository-structure-and-python

File: local.yml (hunk 2 above)
Comment (renceInbox) [reply local.yml]: Added @jeremi

File: filesystems.py (hunk 1 above)
Comment (renceInbox) [reply filesystems.py]: Tried this last time but it is being closed somewhere before we can use it for upload. So when we try to catch the file in `OpenSPPFTPHandler.on_file_received`, it is already deleted. I tried finding the path where it is closed but got stucked in `FTPHandler.ftp_STOR`. I skipped the approach and use tempfile instead to catch up with the demo. Do you want me to revisit this approach?

Comment (renceInbox) [reply filesystems.py]: Added changes for in-memory file

**Comment by renceInbox** [comment]
@jeremi , I added [`tmpfs` in docker config.](https://github.com/OpenSPP/openspp-dms-ftp-server/pull/1/files#diff-342bbbd527cd4be2219251b786ad5d16636b4757e6e6c66bf0dc12dc854c0c13R10)

**Review by jeremi — APPROVED** [review]

**Pull request merged**